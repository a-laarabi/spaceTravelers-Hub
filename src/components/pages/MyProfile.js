import React from 'react';
import { useSelector } from 'react-redux';

function MyProfile() {
  const rockets = useSelector((state) => state.rockets.rockets);
  const missions = useSelector((state) => state.missions);
  const joinedMissions = missions.filter((mission) => mission.reserved);
  const reservedRockets = rockets.map((rocket) => (rocket.reserved ? (
    <li key={rocket.id} className="list-group-item">
      {rocket.name}
    </li>
  ) : false));

  return (
    <div className="container d-flex justify-content-center my-5">
      <div>
        <h2 className="ms-5">My Missions</h2>
        <div className="card rocket-cart ms-5">
          <ul className="list-group list-group-flush">
            {joinedMissions.map((mission) => <li className="p-2 text-danger" key={mission.id}>{mission.name}</li>)}
          </ul>
        </div>
      </div>
      <div>
        <h2 className="ms-5">My Rockets</h2>
        <div className="card rocket-cart ms-5">
          <ul className="list-group list-group-flush">
            {reservedRockets.length > 0 ? reservedRockets : <p className="p-2 text-danger">Please reserve a rocket</p>}
          </ul>
        </div>
      </div>
    </div>
  );
}
export default MyProfile;
