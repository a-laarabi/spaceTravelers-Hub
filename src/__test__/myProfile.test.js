import React from 'react';
import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import { Provider } from 'react-redux';
import store from '../redux/configureStore';
import MyProfile from '../components/pages/MyProfile';

describe('Display missions element', () => {
  test('Check if Profile Component Renders in DOM properly', () => {
    render(
      <Provider store={store}>
        <MyProfile />
      </Provider>,
    );
    expect(screen.queryByText(/My Rockets/)).toBeTruthy();
    expect(screen.queryByText(/My Missions/)).toBeTruthy();
  });
  it('renders properly', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <MyProfile />
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
