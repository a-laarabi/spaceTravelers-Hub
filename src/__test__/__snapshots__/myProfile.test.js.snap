// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Display missions element renders properly 1`] = `
<div
  className="container d-flex justify-content-center my-5"
>
  <div>
    <h2
      className="ms-5"
    >
      My Missions
    </h2>
    <div
      className="card rocket-cart ms-5"
    >
      <ul
        className="list-group list-group-flush"
      />
    </div>
  </div>
  <div>
    <h2
      className="ms-5"
    >
      My Rockets
    </h2>
    <div
      className="card rocket-cart ms-5"
    >
      <ul
        className="list-group list-group-flush"
      >
        <p
          className="p-2 text-danger"
        >
          Please reserve a rocket
        </p>
      </ul>
    </div>
  </div>
</div>
`;
